tests/pos-macros/i4023/Test_2.scala

```diff
@@ -1,3 +1,3 @@
-object Test {
+def test(using quoted.QuoteContext) = {
   Macro.ff(3)
 }
```

Comment:
It's not clear why `using quoted.QuoteContext` is necessary for macro usage.

Reply:
The test is awkward. The inline method is the one that requires the `quoted.Type` as an implicit. This might only be used to write a macro that is used in a macro implementation, hence the QuoteContext.

Previously it worked because we never used the `Type` and by chance it stayed encoded as a `QuoteContext ?=> Type[T]` which was never called. That is why we missed the need for a `QuoteContext`.

The current approach homogeneously requires the quote context for all `Type` quotes as we do with `Expr` quotes. This helps to find out missing QuoteContext early in the pipeline.